src/plugins/data_source_management/public/components/direct_query_data_sources_components/direct_query_data_connection/manage_direct_query_data_connections_table.tsx

```diff
@@ -370,8 +391,103 @@ export const ManageDirectQueryDataConnectionsTable = ({
     ],
   };
 
+  const onDissociate = async (item: DataSourceTableItem) => {
+    const confirmed = await overlays.openConfirm('', {
+      title: i18n.translate('dataSourcesManagement.dataSourcesTable.removeAssociation', {
+        defaultMessage: 'Remove association',
+      }),
+      buttonColor: 'danger',
+    });
+    if (confirmed) {
+      setIsLoading(true);
+      if (workspaceClient && currentWorkspace) {
+        await workspaceClient.dissociate(
+          [{ id: item.id, type: 'data-source' }],
+          currentWorkspace.id
+        );
+      }
+      await fetchDataSources();
+      if (defaultDataSourceId === item.id) {
+        setFirstDataSourceAsDefault(savedObjects.client, uiSettings, true);
+      }
+    }
+  };
+
+  const associateDataSourceButton = DataSourceAssociation && [
+    {
+      renderComponent: (
+        <DataSourceAssociation
+          excludedDataSourceIds={data.map((ds) => ds.id)}
+          onComplete={() => fetchDataSources()}
+        />
+      ),
+    } as TopNavControlComponentData,
+  ];
+
+  const isDashboardAdmin = !!application?.capabilities?.dashboards?.isDashboardAdmin;
+  const canAssociateDataSource =
+    !!currentWorkspace && !currentWorkspace.readonly && isDashboardAdmin;
```

Comment:
Nit: Since we already check `isDashboardAdmin` at the end, do we need to check if workspace readonly here?

Reply:
what's `currentWorkspace.readonly`?

Reply:
Removed